### ISSUE TITLE
User Registration Persistence

### ISSUE THREAD

**Issue by RussellAult** [ZeWaren/jabber-hangouts-transport#5]
I don't know how hard this would be to implement, but it would be great if a user registering with the component would persist across connections. As it currently stands, every time I restart my XMPP client or the component I have to generate a new OAth token and re-register. Re-registering after component restarts is probably a good idea (since it means the OAth tokens don't need to be written into storage), but having to re-register every time my client restarts is a bit annoying (especially since the computer my client is running on isn't quite stable yet).

Thanks!

**Comment by ZeWaren** [comment]
Well, connection persistence IS implemented. If it's not working on your setup, then it's a bug.
You're supposed to only register once. Otherwise, it would indeed be annoying.

Can you check that your spool file and directory are populated?

In my setup:

In the config file:

```
<!-- Where to store the Hangouts OAuth refresh tokens -->
<refreshTokenDirectory>/var/spool/jabberhangouts/refresh_tokens</refreshTokenDirectory>

<!-- Where to store the registered users' information -->
<spoolFile>/var/spool/jabberhangouts/spoolfile</spoolFile>
```

In the system:

```
# ls -l /var/spool/jabberhangouts/refresh_tokens
total 1
-rw-r--r--  1 jabberhangouts  jabberhangouts  66 Dec 10 18:24 b6a58868adab637fbd767874d97145bf55dc02cd

# cat /var/spool/jabberhangouts/refresh_tokens/b6a58868adab637fbd767874d97145bf55dc02cd
1/e9_2XvCpA5ah2SQTsh6zKdKSUdea6rS9ZzOWlGLuR4R80RDknAdJa_sgfheVM5XT

# file /var/spool/jabberhangouts/spoolfile.db
/var/spool/jabberhangouts/spoolfile.db: Berkeley DB 1.85 (Hash, version 2, native byte-order)
```

If nothing is there, maybe the directories don't exist, or maybe the transport is not allowed to write into them (check the UNIX permissions).

**Comment by RussellAult** [comment]
Definitely a "directories don't exist" problem, specifically the `refresh_tokens` directory didn't exist. Interestingly, that appears to be the only `/var` directory that can be missing without causing the component to throw some kind of exception, which is probably why I hadn't noticed it was missing before (e.g. if `/var/spool/jabberhangouts/` is missing, one of the Python libraries throws an exception and quits). It might be nice if a missing `refresh_tokens` directory caused an exception to be thrown too, or at least an warning in the log file.

Thanks!

**Comment by ZeWaren** [comment]
I'll have the component check that the directories are writable at startup, and error if it's not the case.

**Comment by ZeWaren** [comment]
Done.

Commit 0192eb4404feb955e4933b79e4a88637ab555413 closes #5.
